---
title: "MBAL vs. TA comparison"
author: "Katrina Kalantar"
date: '`r format(Sys.Date())`'
output:
  github_document: default
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
library(scales)
library(vegan)
library(knitr)
library(viridis)
library(ggplot2)
library(kableExtra)
options(knitr.table.format = "html") 

generate_pdfs = FALSE
```





**Relevant Definitions:**

**mBAL:** mini broncheolar alveolar lavage

**TA:** tracheal aspirate

**LRTI:** lower respiratory tract infection

**PNA-pos:** Pneumonia positive

**PNA-neg:** Pneumonia negative


**ABSTRACT:** 
Pneumonia causes more deaths each year in the United States than any other type of infectious disease. The ability to accurately detect etiologic pathogens and distinguish them from background commensal microbiota is essential for guiding optimal antimicrobial treatments. In patients requiring mechanical ventilation, less invasive tracheal aspirate sampling (TA) has historically been considered inferior to mini-bronchial alveolar lavage/telescoping catheter (mBAL) specimen collection due to the potential for oropharyngeal microbiota contamination. This idea has been challenged, however, by studies demonstrating a lack of clinically significant differences between sample types, and a greater acceptance of TA sampling is now reflected in recent updates to clinical practice guidelines.4 Despite the broad potential implications of this shift in diagnostic sampling approach, relatively little information exists regarding microbial composition differences between mBAL and TA specimens, and the potential implications of such differences for clinical diagnostic testing. 


***


### Import the microbe data and metadata.

The microbial counts per individual were generated as follows:

Detection of host transcripts and airway microbes leveraged a custom bioinformatics pipeline that incorporated quality filtering using PRICESeqfilter and removal of host sequences by alignment against a database of the human genome (NCBI GRC h38) and Pan troglodytes (UCSC PanTro4) using the STAR aligner. Additional filtering was performed using Bowtie2 to remove non-fungal eukaryotes, cloning vectors and phiX phage. The identities of the remaining microbial reads were determined by querying the NCBI nucleotide (NT) and non-redundant protein (NR) databases using GSNAP-L and RAPSEARCH2, respectively. Microbial alignments detected by RNA-Seq and DNA-Seq were aggregated to the genus-level.

Then, the NT genus rpM counts were merged across all samples. Metadata describing the sample and patient relationships, as well as dividing patients into four groups was imported as .csv.

```{r}
#IMPORT DATA
#.csv of all microbes by TA/mBAL sample ID #012618/
all_microbe_data <- read.csv('./data/020718/BM_4/merged_genusrpm.tsv', sep='\t', header=TRUE, row.names=1)

#read in metadata #012618/
metadata <- read.csv('./data/020718/tavmbal_metadata_noviruses.tsv', sep='\t', row.names=1)
```


The raw microbe data appeared as follows (showing only the first 5 rows): 
```{r, echo = FALSE}
kable(head(all_microbe_data, n=5), "markdown" ) #%>% 
  #kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>%
  #scroll_box(width = "900px", height = "200px")
```
 

And the metadata was:

Note, that effective_group = 1 refers to samples indicated as PNA-pos (unequivocal pneumonia-positive) in the manuscript and effective_group = 4 refers to samples indicated as PNA-neg (unequivocal pneumonia-negative, with clear alternative cause for respiratory failure). "effective_group"'s 2 and 3 are lumped together to form the PNA-unk (pneumonia-unknown) group. 

For more information on the adjudication process, refer to "Cohort Summary" below.
```{r, echo = FALSE}
kable(metadata, "markdown") #%>% 
  #kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>%
  #scroll_box(width = "900px", height = "200px")
```

***

### Cohort Summary 

The initial cohort contained 52 adults 21 to over 89 years of age who were admitted to the intensive care unit with acute respiratory failure. Matched mBAL and TA specimens were obtained from each patient. For analysis purposes, all patients patients were stratified post-collection across four groups according to their likelihood of having a lower respiratory tract infection. 

Two-physician adjudication was used to identify subjects with culture-confirmed bacterial pneumonia (PNA-pos, n = 15) based on retrospective medical record review (blinded to mNGS results) using the United States Centers for Disease Control surveillance case definition of pneumonia.
 
Adjudication is noted in the analysis under the following categories:

Group 1 = **"PNA-pos"**, Definite bacterial pneumonia

**"PNA-unk"**, pneumonia unknown (all patients not in PNA-pos or PNA-neg, grouped together for the purpose of comparing sample types)

Group 2 = Probable Lower Respiratory Tract Infection, but negative clinical microbiology; AND definite viral pneumonia cases (ie pt 205, 228, 295) where infection with RNA virus could not be detected by DNA-seq. 

Group 3 = Unknown LRTI status.

Group 4 = **"PNA-neg"**, Definitely no LRTI, with clear alternative explanation for acute respiratory failure


The distribution of patients is as follows:

```{r, echo = FALSE, fig.height=4, fig.width=6}
barplot(table(metadata$effective_group), col=c("darkred","red","gray40","blue"),
        main = 'Number of Patients per Group', xlab='Group',ylab='# of Patients')
kable(table(metadata$effective_group), "markdown") %>% 
  kable_styling(bootstrap_options = c("striped"), full_width = F, position = "float_right")
```

***

### Comparison of DNA Microbial Burden

#### No differences in Burden of microbial genera between TA and mBAL
```{r}
burden <- colSums(all_microbe_data) 
```

**Microbial burden**, as measured by the total microbial alignments by genus per million reads sequenced (rpM), did not significantly differ between TA and mBAL when using whole genome DNA sequencing: WGS, Wilcoxon rank sum p =
```{r, include=FALSE}
#Difference in burden between G1 TA and G1 mBAL
print(wilcox.test(burden[names(burden) %in% as.character(unlist(metadata[metadata$effective_group %in% c(1),]["TA_DNAseq_filename"]))],
                  burden[names(burden) %in% as.character(unlist(metadata[metadata$effective_group  %in% c(1),]["mBAL_DNAseq_filename"]))])$p.value)

#Difference in burden btwn G4 TA and G4 mBAL
print(wilcox.test(burden[names(burden) %in% as.character(unlist(metadata[metadata$effective_group %in% c(4),]["TA_DNAseq_filename"]))],
                  burden[names(burden) %in% as.character(unlist(metadata[metadata$effective_group  %in% c(4),]["mBAL_DNAseq_filename"]))])$p.value)

#**Evaluating the difference in burden between group 1 and group 4 patients, within sample type** 
#Difference in burden btwn G1 TA and G4 TA
wilcox.test(burden[names(burden) %in% as.character(unlist(metadata[metadata$effective_group %in% c(1),]["TA_DNAseq_filename"]))],
            burden[names(burden) %in% as.character(unlist(metadata[metadata$effective_group  %in% c(4),]["TA_DNAseq_filename"]))])$p.value

#Difference in burden btwn G1 mBAL and G4 mBAL
wilcox.test(burden[names(burden) %in% as.character(unlist(metadata[metadata$effective_group %in% c(1),]["mBAL_DNAseq_filename"]))],
            burden[names(burden) %in% as.character(unlist(metadata[metadata$effective_group  %in% c(4),]["mBAL_DNAseq_filename"]))])$p.value

```

Summary of burden in TA:
```{r, echo=FALSE}
kable(t(as.matrix(summary(burden[names(burden) %in% as.character(unlist(metadata[metadata$effective_group %in% c(1,2,3,4),]["TA_DNAseq_filename"]))]))),"markdown") %>%
  kable_styling(bootstrap_options = c("striped","hover","condensed","responsive"),full_width=F)
```

Summary of burden in mBAL:
```{r, echo=FALSE}
kable(t(as.matrix(summary(burden[names(burden) %in% as.character(unlist(metadata[metadata$effective_group %in% c(1,2,3,4),]["mBAL_DNAseq_filename"]))]))),"markdown") %>%
  kable_styling(bootstrap_options = c("striped","hover","condensed","responsive"),full_width=F)
```

Wilcox rank sum p-value for comparing microbial burden in TA v. mBAL
```{r, echo=FALSE}
# TOTAL BURDEN (all TA v all mBAL)
wilcox.test(burden[names(burden) %in% as.character(unlist(metadata[metadata$effective_group %in% c(1,2,3,4),]["TA_DNAseq_filename"]))],
                  burden[names(burden) %in% as.character(unlist(metadata[metadata$effective_group  %in% c(1,2,3,4),]["mBAL_DNAseq_filename"]))])$p.value[1]
```

***

### Analysis of microbial community proportions

#### Create data splits, normalize microbe counts by Total Sum Scaling, and group all low-level microbes < 1% of total reads into "Other".
Separate MBAL and TA into dataframes, normalize the microbial data into proportions, and separate out group 1.

```{r}
MBAL <- as.character(unname(unlist(metadata[c("mBAL_DNAseq_filename")])))  # separate mBAL from TA
TA <- as.character(unname(unlist(metadata[c("TA_DNAseq_filename")])))
MBAL.microbes <- all_microbe_data[,colnames(all_microbe_data) %in% MBAL]  
TA.microbes <- all_microbe_data[,colnames(all_microbe_data) %in% TA]
MBAL.microbes.norm <-  t(t(MBAL.microbes)/colSums(MBAL.microbes))   # normalize (Total Sum Scaling) -> generate proportions
TA.microbes.norm <- t(t(TA.microbes)/colSums(TA.microbes))
 
# Isolate Group 1 Data
group1.MBAL <- as.character(unname(unlist(metadata[metadata$effective_group==1,][c("mBAL_DNAseq_filename")])))
group1.TA <- as.character(unname(unlist(metadata[metadata$effective_group==1,][c("TA_DNAseq_filename")])))
group1.MBAL.microbes <- all_microbe_data[,colnames(all_microbe_data) %in% group1.MBAL]
group1.MBAL.microbes.norm <- t(t(group1.MBAL.microbes)/colSums(group1.MBAL.microbes))
group1.TA.microbes <- all_microbe_data[,colnames(all_microbe_data) %in% group1.TA]
group1.TA.microbes.norm <- t(t(group1.TA.microbes)/colSums(group1.TA.microbes))

all_microbe_data.microbes.norm <- t(t(all_microbe_data)/colSums(all_microbe_data)) # normalize (TSS)

write.csv(all_microbe_data.microbes.norm, "./output/TableS3.tsv", sep="\t", quote=FALSE) # save normalized data for manuscript prior to filtering microbes with < 1% per sample

all_microbe_data_orig <- all_microbe_data.microbes.norm  # re-assign variable so we have a copy with non-normalized values
all_microbe_data.microbes.norm[all_microbe_data.microbes.norm<.01] <- 0  # remove microbes present at < 1% abundance
new <- rbind(all_microbe_data.microbes.norm, c(1-colSums(all_microbe_data.microbes.norm)))  # add "Other" category
rownames(new) <- c(rownames(all_microbe_data.microbes.norm),"Other")  # adjust row names
all_microbe_data <- new  

```

The normalized data (without removing microbes present at abundance < 1%) is reported in TableS3 of the manuscript, but here we show the per-sample proportions of the top 10 most abundant genera across the cohort after grouping all microbes present at abundance < 1% into the "Other" category:

Note: this is still different from the "Other" values reported in FigureS1, which includes rpm values for all microbes not in the top 39 most abundant microbes.
```{r}
kable(all_microbe_data[names(head(sort(rowSums(all_microbe_data),decreasing=TRUE),n=20)),], "markdown" ) #%>% 
  #kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>%
  #scroll_box(width = "900px", height = "200px")
```

***


### Obtain Community Composition metrics for genus-level microbial proportions

Ultimately, we are interested in genus-level comparisons. Clinical diagnoses are made at the species- and strain-level, but our analysis of NGS data from airway samples has been limited to genus-level comparison due to technical challenges in calling species-level microbial hits (with strong confidence and accuracy). To avoid these issues, we collapse all hits to the genus-level. Thus, comparing the genus-level similarity between samples is important for understanding the impact it would have on ongoing projects.

Iterate over all pairs of mBAL and TA samples from individuals to generate the community similarity metrics at the genus level.
Initial output: a barchart of genera represented in each sample. The bar charts are separated by "group" - group 1 in the first two rows, group 2 in the second...etc.

The main code to accomplish this is shown here:
```{r,fig.width=12,fig.height=6}
generate_pdfs = FALSE

file_pair_order <- c()

all_microbe_data <- all_microbe_data[order(rowSums(all_microbe_data),decreasing=F),]
all_microbe_data <- all_microbe_data[,!is.na(colSums(all_microbe_data))]
mbal_phyla <- all_microbe_data[,colnames(all_microbe_data) %in% MBAL]
ta_phyla <- all_microbe_data[,colnames(all_microbe_data) %in% TA]

#if(generate_pdfs){
#  pdf('./output/genus_analysis.pdf',width=16,height=6)
#}

full_correlations <- vector("list",length=4)
full_bc <- vector("list",length=4)
full_jaccard <- vector("list", length=4)
full_Shannons <- vector("list",length=8) #separate by TA and mBAL for each group
full_richness <- vector("list",length=8)
for(j in c(1,2,3,4)){
  
  correlations <- c()
  bc_dist <- c()
  ta_sdi <- c()
  mbal_sdi <- c()
  names <- c()
  mbal_richness <- c()
  ta_richness <- c()
  jaccard <- c()
  par(mfrow=c(2,9))
  
  for( i in rownames(metadata)){
  mbal_filename <- as.character(metadata[rownames(metadata) == i,]$mBAL_DNAseq_filename)
  ta_filename <- as.character(metadata[rownames(metadata) == i,]$TA_DNAseq_filename)
  
  if(!is.na(metadata[rownames(metadata) == i,]$effective_group)){
    
    if(metadata[rownames(metadata) == i,]$effective_group == j){
    
      # genus barplot
      #barplot(as.matrix(all_microbe_data[,c(mbal_filename,ta_filename)]),col=bp,cex.axis = .6,main=i,names.arg = c("mBAL","TA"))
      file_pair_order <- c(file_pair_order, mbal_filename, ta_filename)
      
      # pairwise pearson correlation - 1 value per patient
      c <- cbind(all_microbe_data[,c(mbal_filename)],all_microbe_data[,c(ta_filename)])
      c <- c[rowSums(c) > 0,]
      colnames(c) <- c(mbal_filename,ta_filename)
      correlations <- c(correlations,cor(c)[2])
      
      # bray-curtis distance - 1 value per patient
      bray_curtis_dist <- vegdist(t(as.matrix(all_microbe_data[,c(mbal_filename,ta_filename)])), method="bray", binary=FALSE, diag=FALSE, upper=TRUE, na.rm = FALSE)
      bc_dist <- c(bc_dist, bray_curtis_dist[1])
      
      # SDI - 1 per sample ( 2 values per patient )
      ta_sdi <- c(ta_sdi,diversity(all_microbe_data[,c(ta_filename)],index="shannon") ) #simpson d
      mbal_sdi <- c(mbal_sdi, diversity(all_microbe_data[,c(mbal_filename)],index="shannon")) #simpson
      
      # jaccard - 1 value per patient
      jaccard_dist <- vegdist(t(as.matrix(all_microbe_data[,c(mbal_filename,ta_filename)])),method = "jaccard",binary = FALSE,diag = FALSE,upper = TRUE,na.rm = FALSE)
      jaccard <- c(jaccard, jaccard_dist[1])
      
      # richness - 1 value per sample ( 2 values per patient )
      mbal_richness <- c(mbal_richness, length(all_microbe_data[,c(mbal_filename)][all_microbe_data[,c(mbal_filename)]>0]))
      ta_richness <- c(ta_richness, length(all_microbe_data[,c(ta_filename)][all_microbe_data[,c(ta_filename)]>0]))
      
      names <- c(names,rownames(metadata[rownames(metadata) == i,]))
    }  
  }
}
full_correlations[[j]] <- correlations
full_bc[[j]] <- bc_dist
full_jaccard[[j]] <- jaccard

full_Shannons[[j + (j-1)]] <- mbal_sdi
full_Shannons[[j+1+(j-1)]] <- ta_sdi

full_richness[[j + (j-1)]] <- mbal_richness
full_richness[[j+1+(j-1)]] <- ta_richness

}
```


```{r, echo = FALSE}

# if(generate_pdfs){
#   dev.off()
# }
# 
# if(generate_pdfs){
#   pdf('./output/genus_analysis.legend.pdf',height=20)
#   barplot(as.matrix(all_microbe_data[,c(mbal_filename,ta_filename)]),col=bp,cex.axis = .6,main=i,names.arg = c("mBAL","TA"),legend=TRUE)
# }

```

### Genus-level sample composition barcharts

Set colors to be used in genus-abundance barplots
```{r}
bp <- c('aquamarine3','bisque3','blue','blue4','darkolivegreen3','darkred','darksalmon','darkslateblue',
        'darkslategray','blueviolet','brown','brown2','darkgoldenrod','darkgoldenrod1','royalblue1',
        'chocolate1','coral2','chartreuse4','chartreuse','cadetblue3','darkkhaki','darkmagenta','deepskyblue',
        'darkorchid1','darkorange1','greenyellow','forestgreen','dodgerblue1','firebrick2','gold1',
        'deeppink','deeppink3','mediumpurple','cornsilk3','cornsilk4','cyan','bisque4','gray39','black','hotpink4')
```

Create a genus-level barplot for each sample. Since we have limited the color palette to only 40 colors, we want anything that isn't in the length of the list of colors (top 39 most abundant microbes in the cohort) to be included in the "Other" category. The "Other" category also includes rpm values for any microbes that are present at < 1% of that particular sample (even if they are of a genera that is in the top 39 most abundant)
```{r,fig.width=12,fig.height=6}
generate_pdfs = FALSE #TRUE

file_pair_order <- c()
bp <- bp[shuffle(bp)]

all_microbe_data <- all_microbe_data[order(rowSums(all_microbe_data),decreasing=F),]
all_microbe_data <- all_microbe_data[,!is.na(colSums(all_microbe_data))]
mbal_phyla <- all_microbe_data[,colnames(all_microbe_data) %in% MBAL]
ta_phyla <- all_microbe_data[,colnames(all_microbe_data) %in% TA]

if(generate_pdfs){
  pdf('./output/genus_analysis.pdf',width=16,height=6)
}

for(j in c(1,2,3,4)){
  
  par(mfrow=c(2,9))
  
  for( i in rownames(metadata)){
  mbal_filename <- as.character(metadata[rownames(metadata) == i,]$mBAL_DNAseq_filename)
  ta_filename <- as.character(metadata[rownames(metadata) == i,]$TA_DNAseq_filename)
  
  if(!is.na(metadata[rownames(metadata) == i,]$effective_group)){
    
    if(metadata[rownames(metadata) == i,]$effective_group == j){
    
      # genus barplot
      
      # get the TA and MBAL sample into a subset of the original matrix
      barplot_sub <- as.matrix(all_microbe_data[,c(mbal_filename,ta_filename)])  
      
      # add all microbes not in the top 39 most abundant to the "Other" variable
      barplot_sub[c("Other"),] <- barplot_sub[c("Other"),] + colSums(barplot_sub[!(rownames(barplot_sub) %in% names(head(sort(rowSums(all_microbe_data), decreasing=TRUE),length(bp)))),])
      
      # subset to the top 40 (39 microbes + 1 Other)
      barplot_sub <- barplot_sub[rownames(barplot_sub) %in% names(head(sort(rowSums(all_microbe_data), decreasing=TRUE),length(bp))),]
      
      # plot it
      barplot(barplot_sub,col=bp,cex.axis = .6,main=i,names.arg = c("mBAL","TA"))

      file_pair_order <- c(file_pair_order, mbal_filename, ta_filename)
      
    }  
  }
}
}
```

```{r}
if(generate_pdfs){
  dev.off()
}

if(generate_pdfs){
  pdf('./output/genus_analysis.legend.pdf',height=20)
  barplot(as.matrix(all_microbe_data[rownames(all_microbe_data) %in% names(head(sort(rowSums(all_microbe_data), decreasing=TRUE),length(bp))),c(mbal_filename,ta_filename)]),col=bp,cex.axis = .6,main=i,names.arg = c("mBAL","TA"),legend=TRUE)
}
```


```{r, fig.width=10, include=FALSE}
library(RColorBrewer)
pdf('./output/genus_analysis_v2.pdf',width=50, height=4)
barplot(all_microbe_data[,file_pair_order],col=brewer.pal(n = 30, name = "Paired"), las=2, cex.names=.4)
dev.off()
pdf('./output/genus_analysis_v2L.pdf',width=80, height=20)
barplot(all_microbe_data[,file_pair_order],col=brewer.pal(n = 30, name = "Paired"), las=2, cex.names=.4,legend=TRUE)
dev.off()
```


***

### Pearson Correlation of all microbes, regardless of patient ID.

We plot the microbial genus rpM from mini-BAL samples versus the microbial genus rpM from Tracheal Aspriate samples and evaluate the pearson correlation between microbe counts across all patients (black). Then, repeat the ananlysis restricting to only the microbes identified in PNA-pos (red) or PNA-neg (blue) patients.

```{r, echo = FALSE}
#filenames <- metadata[,c("mBAL_DNAseq_filename","TA_DNAseq_filename")]
reds <- rgb(colorRamp(c("red", "red3", "red4"), space="rgb", interpolate="linear")(0:255/255), maxColorValue=255)  
 #palette(brewer.pal(n = 20, name = "Reds"))
blues <- rgb(colorRamp(c("blue", "blue2", "blue3"), space="rgb", interpolate="linear")(0:255/255), maxColorValue=255)  #palette(brewer.pal(n = 20, name = "Blues")) #"blue3", "blue2",  'blue', "white",  

microbes_TA_g1 <- c()
microbes_MBAL_g1 <- c()
microbes_TA_g4 <- c()
microbes_MBAL_g4 <- c()
microbes_TA_g2 <- c()
microbes_MBAL_g2 <- c()

microbes_TA <- c()
microbes_MBAL <- c()

metadata[is.na(metadata)] <- 3

for(i in seq(1:nrow(metadata))){
  colval <- 'grey'
  #print(metadata[i,c("effective_group")])
  
  #print(metadata[i,"mBAL_DNAseq_filename"])
  #print(metadata[i,"TA_DNAseq_filename"])
  #print(all_microbe_data[,metadata[i,c("mBAL_DNAseq_filename")]][all_microbe_data[,metadata[i,c("mBAL_DNAseq_filename")]]>0])
  #print(all_microbe_data[,metadata[i,c("TA_DNAseq_filename")]][all_microbe_data[,metadata[i,c("TA_DNAseq_filename")]] > 0])
  plot_data <- cbind(all_microbe_data[,as.character(metadata[i,c("mBAL_DNAseq_filename")])],all_microbe_data[,as.character(metadata[i,c("TA_DNAseq_filename")])])
  #plot_data <- cbind(log10(all_microbe_data[,as.character(metadata[i,c("mBAL_DNAseq_filename")])]*100),log10(all_microbe_data[,as.character(metadata[i,c("TA_DNAseq_filename")])]*100))
  colnames(plot_data) <- c("MBAL","TA")
  plot_data <- plot_data[rowSums(plot_data) > 0,]
  
  if(metadata[i,c("effective_group")] == "1"){
    #colval <- 'red'
    colval <- reds[i*2]
    #print(colval)
    if(class(plot_data)=="numeric"){
        microbes_TA_g1 <- c(microbes_TA_g1,plot_data[c("TA")])
        microbes_MBAL_g1 <- c(microbes_MBAL_g1,plot_data[c("MBAL")])
      }else{
        microbes_TA_g1 <- c(microbes_TA_g1,plot_data[,c("TA")])
        microbes_MBAL_g1 <- c(microbes_MBAL_g1,plot_data[,c("MBAL")])
      }
  }else if(metadata[i,c("effective_group")] == "4"){
    #colval <- 'blue'
    colval <- blues[i*2]
    #print(colval)
    if(class(plot_data)=="numeric"){
        microbes_TA_g4 <- c(microbes_TA_g4,plot_data[c("TA")])
        microbes_MBAL_g4 <- c(microbes_MBAL_g4,plot_data[c("MBAL")])
      }else{
        microbes_TA_g4 <- c(microbes_TA_g4,plot_data[,c("TA")])
        microbes_MBAL_g4 <- c(microbes_MBAL_g4,plot_data[,c("MBAL")])
      }
  }else if(metadata[i,c("effective_group")] == "2"){
    #colval <- 'blue'
    colval <- blues[i*2]
    #print(colval)
    if(class(plot_data)=="numeric"){
        microbes_TA_g2 <- c(microbes_TA_g2,plot_data[c("TA")])
        microbes_MBAL_g2 <- c(microbes_MBAL_g2,plot_data[c("MBAL")])
      }else{
        microbes_TA_g2 <- c(microbes_TA_g2,plot_data[,c("TA")])
        microbes_MBAL_g2 <- c(microbes_MBAL_g2,plot_data[,c("MBAL")])
      }
  }
  if(class(plot_data)=="numeric"){
    microbes_TA <- c(microbes_TA,plot_data[c("TA")])
    microbes_MBAL <- c(microbes_MBAL,plot_data[c("MBAL")])
  }else{
    microbes_TA <- c(microbes_TA,plot_data[,c("TA")])
    microbes_MBAL <- c(microbes_MBAL,plot_data[,c("MBAL")])
  }
}

```


```{r, echo = FALSE, fig.height=3, fig.width=3 }

if(generate_pdfs){
pdf('./output/genus.pearson.pdf',height=5, width=5)
}
par(mfrow=c(1,3))

#postscript(file = './output/genus.pearson3.ps',onefile=TRUE,height=5, width=5)
gALL<- as.data.frame(cbind(microbes_TA,microbes_MBAL))
scatter_plot2 <- ggplot(as.data.frame(gALL), aes(microbes_TA, microbes_MBAL))
scatter_plot2 + geom_point(shape=3,colour="black") + labs(x = "mini-BAL", y = "Tracheal Aspirate", cex=.8)  + geom_smooth(method="lm",colour="black") + xlim(0,1) + ylim(0,1)+theme(axis.text=element_text(size=8),
        axis.title=element_text(size=10)) + ggtitle("All Microbes")
#dev.off()

#postscript(file = './output/genus.pearson.ps',onefile=TRUE,height=5, width=5)
g1 <- as.data.frame(cbind(microbes_TA_g1,microbes_MBAL_g1))
scatter_plot1 <- ggplot(as.data.frame(g1), aes(microbes_TA_g1, microbes_MBAL_g1))
scatter_plot1 + geom_point(shape=3,colour="red") +labs(x = "mini-BAL", y = "Tracheal Aspirate", cex=.8)  + geom_smooth(method="lm",colour="red") + xlim(0,1) + ylim(0,1)+theme(axis.text=element_text(size=8),
        axis.title=element_text(size=10))+ ggtitle("Microbes from PNA-pos pts")
#dev.off()

#postscript(file = './output/genus.pearson2.ps',onefile=TRUE,height=5, width=5)
g4<- as.data.frame(cbind(microbes_TA_g4,microbes_MBAL_g4))
scatter_plot2 <- ggplot(as.data.frame(g4), aes(microbes_TA_g4, microbes_MBAL_g4))
scatter_plot2 + geom_point(shape=3,colour="blue") + labs(x = "mini-BAL", y = "Tracheal Aspirate", cex=.8)  + geom_smooth(method="lm",colour="blue") + xlim(0,1) + ylim(0,1)+theme(axis.text=element_text(size=8),
        axis.title=element_text(size=10))+ ggtitle("Microbes from PNA-neg pts")
```

Pearson correlation for all microbes was:
```{r, echo = FALSE}
cor.test(gALL$microbes_TA, gALL$microbes_MBAL, method = "pearson", conf.level = 0.95)
```

Pearson correlation for microbes found in PNA-pos patients was:
```{r, echo = FALSE}
cor.test(g1$microbes_TA_g1, g1$microbes_MBAL_g1, method = "pearson", conf.level = 0.95)
```

Pearson correlation for microbes found in PNA-neg patients was:
```{r, echo = FALSE}
cor.test(g4$microbes_TA_g4, g4$microbes_MBAL_g4, method = "pearson", conf.level = 0.95)

if(generate_pdfs){
  dev.off()
}

```


```{r, include = FALSE}
### EVALUATE WHETHER THE PEARSON CORR for G1 is statistically significantly larger than the corr for G4

library(psych)
r1 <- cor.test(g1$microbes_TA_g1, g1$microbes_MBAL_g1, method = "pearson", conf.level = 0.95)$estimate
r2 <- cor.test(g4$microbes_TA_g4, g4$microbes_MBAL_g4, method = "pearson", conf.level = 0.95)$estimate

# function to do fisher transformations 
fisher.z<- function (r1,r2,n1,n2) ((0.5*log((1+r1)/(1-r1)))-(0.5*log((1+r2)/(1-r2))))/((1/(n1-3))+(1/(n2-3)))^0.5

# or this (either version will suffice) 
fisher.z<- function(r1,r2,n1,n2) (atanh(r1) - atanh(r2)) / ((1/(n1-3))+(1/(n2-3)))^0.5

#input n and r from correlations manually (two tailed test)
2*(1-pnorm(abs(fisher.z(r1,r2,n1= length(g1$microbes_TA_g1),n2= length(g4$microbes_TA_g4)))))

diff.corr <- function( r1, n1, r2, n2 ){ 

    Z1 <- 0.5 * log( (1+r1)/(1-r1) ) 
    Z2 <- 0.5 * log( (1+r2)/(1-r2) ) 

    diff   <- Z1 - Z2 
    SEdiff <- sqrt( 1/(n1 - 3) + 1/(n2 - 3) ) 
    diff.Z  <- diff/SEdiff 

    p <- 2*pnorm( abs(diff.Z), lower=F) 
    cat( "Two-tailed p-value", p , "\n" ) 
  } 

  diff.corr( r1, n1=length(g1$microbes_MBAL_g1), r2, n2=length(g4$microbes_MBAL_g4) ) 
  ## Two-tailed p-value 0.4103526 

  diff.corr( r1=0.1, n1=100, r2=-0.1, n2=80 ) 
  ## Two-tailed p-value 0.1885966 

```

***

### Per-patient Pearson Correlation
We evaluate the pearson correlation between mBAL and TA samples from the same individual. Here we obtain a single correlation value per patient. Then, we evaluate the difference in pearson correlation values for patients in PNA-pos versus PNA-neg patients.

Note: this analysis was ultimately not included in the manuscript, however is consistent with the conclusions of the above analysis of "microbe pearson correlation, regardless of patient ID" - indicating that correlation is stronger between TA and mBAL sampels in PNA-pos patients, likely drivent by high abundance of few microbes in the context of an infection.

```{r, results="hide", include=FALSE}
if(generate_pdfs){
  pdf('./output/genus.pearson.pdf',height=5, width=5)
}
names(full_correlations) <- c("Group 1","Group 2", "Group 3", "Group 4")
stripchart(full_correlations,method='jitter',jitter=.1,vertical=T,col='navyblue',pch=16,cex=2,main="Pearson Correlation by Patient Group",xlab="Patient Group", ylab="Pearson Correlation")
# dev.off()

#Overall correlation is variable
print(paste("mean pearson correlation over all group: ",mean(unlist(full_correlations))))
print(paste("sd pearson corr over all groups: ",sd(unlist(full_correlations))))

#But correlation in group 1 is much higher than correlation in group 4
print(paste("mean pearson correlation over group1: ",mean(full_correlations[[1]])))
print(paste("sd pearson corr over group 1: ",sd(full_correlations[[1]])))
print(paste("mean pearson correlation over group2: ",mean(full_correlations[[2]])))
print(paste("sd pearson corr over group 2: ",sd(full_correlations[[2]])))
print(paste("mean pearson correlation over group3: ",mean(full_correlations[[3]])))
print(paste("sd pearson corr over group 3: ",sd(full_correlations[[3]])))
print(paste("mean pearson correlation over group4: ",mean(full_correlations[[4]])))
print(paste("sd pearson corr over group 4: ",sd(full_correlations[[4]])))
```

Summary of Per-patient, Pearson Correlation between mBAL and TA:
```{r}
#print(summary(unlist(full_correlations)))
kable(t(as.matrix(summary(unlist(full_correlations)))), "markdown") %>%
  kable_styling(bootstrap_options = c("striped","hover","condensed","responsive"),full_width=F)#,position = "float_right")
```

We note that there is a difference in the genus-level pearson correlation between Group 1 and Group 4 individuals.
```{r}
wilcox.test(full_correlations[[1]],full_correlations[[4]])
```


***

### Genus-level Bray-Curtis Distance: 

We evaluate the Bray-curtis community similarity metric between mBAL and TA samples from the same individual, and evaluate significance using the PERMANOVA test.

For reference, PERMANOVA (Permutational multivariate analysis of variance), as definted by Wikipedia is: a non-parametric multivariate statistical test. PERMANOVA is used to compare groups of objects and test the null hypothesis that the centroids and dispersion of the groups as defined by measure space are equivalent for all groups. A rejection of the null hypothesis means that either the centroid and/or the spread of the objects is different between the groups. Hence the test is based on the prior calculation of the distance between any two objects included to the experiment.

In studies ecology studies, the Bray-Curtis community similarity metric is a beta diversity metirc used to quantify the dissimilarity between two different sites based on counts of organisms at each site. In this case, we evaluate TA and mBAL as two distinct sites and use Bray-Curtis distance as the input to the PERMANOVA analysis.

```{r, include = FALSE}
#if(generate_pdfs){
#  pdf('./output/genus.bc.pdf',height=5, width=5)
#}
# there is a significant difference in the genus-level BC dist between infxn and no infxn
#names(full_bc) <- c("Group 1","Group 2", "Group 3", "Group 4")
#stripchart(full_bc,method='jitter',jitter=.1,vertical=T,col=alpha('navyblue',.6),pch=16,cex=2,main="Bray Curtis Similarity by Patient Group",xlab="Patient Group", ylab="Bray Curtis Similarity")
# dev.off()

#Overall Bray-Curtis is variable
print(paste("mean BC over all group: ",mean(unlist(full_bc))))
print(paste("sd BC over all groups: ",sd(unlist(full_bc))))

#But Bray-Curtis in group 1 is much higher than correlation in group 4
print(paste("mean BC over group1: ",mean(full_bc[[1]])))
print(paste("sd BC over group1 : ",sd(full_bc[[1]])))
print(paste("mean bray curtis over group2: ",mean(full_bc[[2]])))
print(paste("sd pearson corr over group 2: ",sd(full_bc[[2]])))
print(paste("mean bray curtis over group3: ",mean(full_bc[[3]])))
print(paste("sd pearson corr over group 3: ",sd(full_bc[[3]])))
print(paste("mean bray curtis over group4: ",mean(full_bc[[4]])))
print(paste("sd pearson corr over group 4: ",sd(full_bc[[4]])))

#There is a significant difference in the genus-level bray-curtis dist between group 1 and group 4
wilcox.test(full_bc[[1]],full_bc[[4]])

#Summary of bray curtis 
print(summary(unlist(full_bc)))
```



**PERMANOVA Analysis**
```{r, echo = FALSE, message = FALSE, results = FALSE, fig.height=4, fig.width=10}

output <- c("./output/")

par(mfrow=c(1,3))

set.seed(5)

BC <- as.matrix(vegan::vegdist(t(all_microbe_data),method="bray"))#as.matrix()
example_NMDS <- vegan::metaMDS(BC,k=2,trymax=1000)
# plot solution 
x <- example_NMDS$points[,1]
y <- example_NMDS$points[,2]
names(x) %in% metadata$mBAL_DNAseq_filename

#Here is an example of how to run a permanova test using the adonis function in vegan. In this example we are testing the hypothesis that the three stations we collected samples from have different centroids
a <-(as.integer(names(x) %in% metadata$mBAL_DNAseq_filename)+1)
names(a) <- names(x)
a <- as.data.frame(a)
colnames(a) <- c("group")
a <- cbind(a,a)#,rownames(a),rownames(a))
colnames(a) <- c("group","x")#,"category","BC")
a$group <- as.factor(a$group)
head(a)
permanova_result_ALL <- vegan::adonis2(BC~group, a, permutations = 999)


if(generate_pdfs){
pdf(paste(output,"BrayCurtisPermanova.pdf"),height=6, width=6) 
}
plot(x, y, xlab="Coordinate 1", ylab="Coordinate 2", 
  main="NMDS of ALL mBAL and TA Samples\nusing Bray-Curtis Dissimilarity",col=c(rgb(244,73,19,maxColorValue = 255),rgb(51,60,163,maxColorValue = 255))[as.integer(names(x) %in% metadata$mBAL_DNAseq_filename) + 1], pch=c(17,15)[as.integer(names(x) %in% c(as.character(metadata[metadata$effective_group==1,c("mBAL_DNAseq_filename")]),as.character(metadata[metadata$effective_group==1,c("TA_DNAseq_filename")]))
) + 1],font.main = 1, cex.axis=.8,cex.main=.8)
vegan::ordispider(example_NMDS, group=a$group, show="1", col=rgb(244,73,19,maxColorValue = 255), kind="se",conf=0.95)
vegan::ordispider(example_NMDS, group=a$group, show="2", col=rgb(51,60,163,maxColorValue = 255), kind="se",conf=0.95)
#text(-.2,.4,"p > .1",cex=.8)
text(-.2,.23,"mini-BAL",cex=.6,col=rgb(51,60,163,maxColorValue = 255))
text(.3,-.23,"Tracheal Aspriate",cex=.6, col=rgb(244,73,19,maxColorValue = 255))
text(x, y, labels = sapply(strsplit(names(x),"[.]"),"[",2), cex=.5)
if(generate_pdfs){
dev.off()
}



# Bray-curtis for just the group 1 samples

set.seed(5)
g1_only <- metadata[metadata$effective_group==1,]

BC <- as.matrix(vegan::vegdist(t(all_microbe_data[,colnames(all_microbe_data) %in% as.character(unlist(g1_only[,c("mBAL_DNAseq_filename","TA_DNAseq_filename")]))]),method="bray"))#as.matrix()
example_NMDS <- vegan::metaMDS(BC,k=2,trymax=1000)
# plot solution 
x <- example_NMDS$points[,1]
y <- example_NMDS$points[,2]
names(x) %in% metadata$mBAL_DNAseq_filename
#plot(x, y, xlab="Coordinate 1", ylab="Coordinate 2", 
#  main="Metric	MDS", col=alpha(c("green2","grey29")[as.integer(names(x) %in% metadata$mBAL_DNAseq_filename) + 1],.5), pch=16)#,	type="n")
#text(x, y, labels = sapply(strsplit(names(x),"[.]"),"[",2), cex=.5)

#Here is an example of how to run a permanova test using the adonis function in vegan. In this example we are testing the hypothesis that the three stations we collected samples from have different centroids
a <-(as.integer(names(x) %in% metadata$mBAL_DNAseq_filename)+1)
names(a) <- names(x)
a <- as.data.frame(a)
colnames(a) <- c("group")
a <- cbind(a,a)#,rownames(a),rownames(a))
colnames(a) <- c("group","x")#,"category","BC")
a$group <- as.factor(a$group)
head(a)
permanova_result_g1 <- vegan::adonis2(BC~group, a, permutations = 999)



if(generate_pdfs){
pdf(paste(output,"BrayCurtisPermanova_g1.pdf"),height=6, width=6) 
}
#plot(x, y, xlab="Coordinate 1", ylab="Coordinate 2", 
#  main="NMDS of mini-BAL and Tracheal Aspirate Samples\nusing Bray-Curtis Dissimilarity", col=alpha(c("green2","grey29")[as.integer(names(x) %in% metadata$mBAL_DNAseq_filename) + 1],.5), pch=16,font.main = 1, #cex.axis=.8,cex.main=.8)
plot(x, y, xlab="Coordinate 1", ylab="Coordinate 2", 
  main="NMDS of PNA-pos mBAL and TA Samples\nusing Bray-Curtis Dissimilarity", col=c(rgb(244,73,19,maxColorValue = 255),rgb(51,60,163,maxColorValue = 255))[as.integer(names(x) %in% metadata$mBAL_DNAseq_filename) + 1], pch=17,font.main = 1, cex.axis=.8,cex.main=.8, cex=1.2)
vegan::ordispider(example_NMDS, group=a$group, show="1", col=rgb(244,73,19,maxColorValue = 255), kind="se",conf=0.95)
vegan::ordispider(example_NMDS, group=a$group, show="2", col=rgb(51,60,163,maxColorValue = 255), kind="se",conf=0.95)
#text(-.2,.4,"p > .1",cex=.8)
text(-.2,.23,"mini-BAL",cex=.6,col=rgb(51,60,163,maxColorValue = 255))
text(.3,-.23,"Tracheal Aspriate",cex=.6, col=rgb(244,73,19,maxColorValue = 255))
text(x, y, labels = sapply(strsplit(names(x),"[.]"),"[",2), cex=.5)

if(generate_pdfs){
dev.off()
}



# Bray-curtis for just the group 4 samples

set.seed(5)
g4_only <- metadata[metadata$effective_group==4,]

BC <- as.matrix(vegan::vegdist(t(all_microbe_data[,colnames(all_microbe_data) %in% as.character(unlist(g4_only[,c("mBAL_DNAseq_filename","TA_DNAseq_filename")]))]),method="bray"))#as.matrix()
example_NMDS <- vegan::metaMDS(BC,k=2,trymax=1000)
# plot solution 
x <- example_NMDS$points[,1]
y <- example_NMDS$points[,2]
names(x) %in% metadata$mBAL_DNAseq_filename

#Here is an example of how to run a permanova test using the adonis function in vegan. In this example we are testing the hypothesis that the three stations we collected samples from have different centroids
a <-(as.integer(names(x) %in% metadata$mBAL_DNAseq_filename)+1)
names(a) <- names(x)
a <- as.data.frame(a)
colnames(a) <- c("group")
a <- cbind(a,a)
colnames(a) <- c("group","x")
a$group <- as.factor(a$group)
permanova_result_g4 <- vegan::adonis2(BC~group, a, permutations = 999)


if(generate_pdfs){
pdf(paste(output,"BrayCurtisPermanova_g4.pdf"),height=6, width=6) 
}
plot(x, y, xlab="Coordinate 1", ylab="Coordinate 2", 
  main="NMDS of PNA-neg mBAL and TA Samples\nusing Bray-Curtis Dissimilarity",  col=c(rgb(244,73,19,maxColorValue = 255),rgb(51,60,163,maxColorValue = 255))[as.integer(names(x) %in% metadata$mBAL_DNAseq_filename) + 1], pch=15,font.main = 1, cex.axis=.8,cex.main=.8, cex=1.2)
vegan::ordispider(example_NMDS, group=a$group, show="1", col=rgb(244,73,19,maxColorValue = 255), kind="se",conf=0.95)
vegan::ordispider(example_NMDS, group=a$group, show="2", col=rgb(51,60,163,maxColorValue = 255), kind="se",conf=0.95)
#text(-.2,.4,"p > .1",cex=.8)
text(x, y, labels = sapply(strsplit(names(x),"[.]"),"[",2), cex=.5)
text(-.2,.23,"mini-BAL",cex=.6,col=rgb(51,60,163,maxColorValue = 255))
text(.3,-.23,"Tracheal Aspriate",cex=.6, col=rgb(244,73,19,maxColorValue = 255))
if(generate_pdfs){
dev.off()
}

```


PERMANOVA result across all samples:
```{r}
print(permanova_result_ALL)
```

PERMANOVA result across only the PNA-pos samples:
```{r}
print(permanova_result_g1)
```

PERMANOVA result across only the PNA-neg samples:
```{r}
print(permanova_result_g4)
```


While no split (all data, PNA-pos only, or PNA-neg only) achieves statistical significantly differences in beta diversity, we note that the PNA-negative samples trend towards more distinct beta diversity in mBAL v. TA samples. Meanwhile, PNA-positive samples show a much more similar beta diversity.


```{r, include = FALSE}
#**GENUS-LEVEL JACCARD SIMILARITY** : We evalaute the genus-level jaccad similarity between sample types from the same individual.

if(generate_pdfs){
  pdf('./output/genus.jaccard.pdf',height=5, width=5)
}
# there is a significant difference in the genus-level jaccard dist between infxn and no infxn
names(full_jaccard) <- c("Group 1","Group 2", "Group 3", "Group 4")
stripchart(full_jaccard,method='jitter',jitter=.1,vertical=T,col='navyblue',pch=16,cex=2,main="Jaccard Similarity by Patient Group",xlab="Patient Group", ylab="Jaccard Similarity")
# dev.off()

#Overall jaccard is variable
print(paste("mean jaccard over all group: ",mean(unlist(full_jaccard))))
print(paste("sd jaccard over all groups: ",sd(unlist(full_jaccard))))

#But jaccard in group 1 is much higher than correlation in group 4
print(paste("mean jaccard over group1: ",mean(full_jaccard[[1]])))
print(paste("sd jaccard over group1 : ",sd(full_jaccard[[1]])))
print(paste("mean jaccard over group2: ",mean(full_jaccard[[2]])))
print(paste("sd jaccard over group 2: ",sd(full_jaccard[[2]])))
print(paste("mean jaccard over group3: ",mean(full_jaccard[[3]])))
print(paste("sd jaccard over group 3: ",sd(full_jaccard[[3]])))
print(paste("mean jaccard over group4: ",mean(full_jaccard[[4]])))
print(paste("sd jaccard over group 4: ",sd(full_jaccard[[4]])))

#Difference in jaccard similarity between Group 1 and Group 4 individuals:
wilcox.test(full_jaccard[[1]],full_jaccard[[4]])

#Summary of jaccard
print(summary(unlist(full_jaccard)))
```

***

### Shannons diversity index

Reduced alpha diversity of the human respiratory microbiome has been described as an ecological marker of infection, and thus we next asked whether Shannon's Diversity Index (SDI) differed by specimen type. 

Here we evaluate the Shannons diversity index (SDI, alpha diversity) for each sample type from each individual. We expect that SDI will be similar across sample type for the same patient.

```{r, include= FALSE}
#Difference in the genus-level Shannons diversity comparing TA to mBAL in any group

print(wilcox.test(full_Shannons[[1]],full_Shannons[[2]])) # dif btwn TA and mBAL SDI in group 1 
print(wilcox.test(full_Shannons[[3]],full_Shannons[[4]])) # dif btwn TA and mBAL SDI in group 2
print(wilcox.test(full_Shannons[[5]],full_Shannons[[6]])) # dif btwn TA and mBAL SDI in group 3
print(wilcox.test(full_Shannons[[7]],full_Shannons[[8]])) # dif btwn TA and mBAL SDI in group 4
print(wilcox.test(c(full_Shannons[[1]],full_Shannons[[3]],full_Shannons[[5]],full_Shannons[[7]]),
                  c(full_Shannons[[2]],full_Shannons[[4]],full_Shannons[[6]],full_Shannons[[8]]))) # dif btwn TA and mBAL SDI across ALL Groups
```

Difference in genus-level Shannons diversity comparing G1 to G4 in mBAL
```{r, echo=FALSE}
print(wilcox.test(full_Shannons[[1]],full_Shannons[[7]]))
```

Significance in genus-level Shannons diversity between G1 and G4 in TA
```{r, echo=FALSE}
print(wilcox.test(full_Shannons[[2]],full_Shannons[[8]]))
```

Summary of Shannons Diversity in mBAL
```{r, echo = FALSE}
kable(t(as.matrix(summary(c(full_Shannons[[1]],full_Shannons[[3]],full_Shannons[[5]],full_Shannons[[7]])))),"markdown") %>%
  kable_styling(bootstrap_options = c("striped","hover","condensed","responsive"),full_width=F)
```

Summary of Shannons Diversity in TA
```{r, echo = FALSE}
kable(t(as.matrix(summary(c(full_Shannons[[2]],full_Shannons[[4]],full_Shannons[[6]],full_Shannons[[8]])))),"markdown") %>%
  kable_styling(bootstrap_options = c("striped","hover","condensed","responsive"),full_width=F)
```

```{r, results="hide", fig.width=8, fig.height=4}
if(generate_pdfs){
  pdf('./output/genus.Shannons.pdf',height=5, width=5)
}
names(full_Shannons) <- c("Group 1\nMBAL","Group 1\nTA", "Group 2\nMBAL","Group 2\nTA", "Group 3\nMBAL","Group 3\nTA", "Group 4\nMBAL","Group 4\nTA")
boxplot(full_Shannons,cex.axis=.5,main="Shannons Diversity by Sample Type and Patient Group",xlab="Patient Group", ylab="Shannons Diversity Index",cex.main=.8)
stripchart(full_Shannons,method='jitter',jitter=.1,vertical=T,col='navyblue',pch=16,cex=1,add=T)
# dev.off()


if(generate_pdfs){
  pdf('./output/genus.shannon_g1g4.pdf',height=5, width=5)
}
f <- full_Shannons[c(1,7,2,8)]
names(f) <- c("Group 1\nMBAL","Group 4\nMBAL","Group 1\nTA", "Group 4\nTA")
boxplot(f,cex.axis=.5,main="Shannon Diversity by Sample Type and Patient Group",xlab="Patient Group", ylab="Shannon Diversity Index",cex.main=.8)
stripchart(f,method='jitter',jitter=.1,vertical=T,col=c('red','navyblue','red','navyblue'),pch=17,cex=1,add=T)
# dev.off()
```

```{r, echo=FALSE, include=FALSE}
print("mean/sd SDI mBAL Group 1:")
print(mean(unlist(full_Shannons[[1]])))
print(sd(unlist(full_Shannons[[1]])))
print("mean/sd SDI TA Group 1:")
print(mean(unlist(full_Shannons[[2]])))
print(sd(unlist(full_Shannons[[2]])))
print("mean/sd SDI mBAL Group 4:")
print(mean(unlist(full_Shannons[[7]])))
print(sd(unlist(full_Shannons[[7]])))
print("mean/sd SDI TA Group 4:")
print(mean(unlist(full_Shannons[[8]])))
print(sd(unlist(full_Shannons[[8]])))
```

***

### Genus-level richness 
We evaluate the genus richness per sample type per patient, expecting no difference in richness between sample types from the same individual.

```{r, echo = FALSE, include=FALSE}
#There is no significant difference in the genus-level richness comparing TA to mBAL in any single group
print(wilcox.test(full_richness[[1]],full_richness[[2]])) # Richness TA v. mBAL Group 1
print(wilcox.test(full_richness[[3]],full_richness[[4]])) # Richness TA v. mBAL Group 2
print(wilcox.test(full_richness[[5]],full_richness[[6]])) # Richness TA v. mBAL Group 3
print(wilcox.test(full_richness[[7]],full_richness[[8]])) # Richness TA v. mBAL Group 4

#But, there is significant difference in the richness when you combine across groups!!!
print(wilcox.test(c(full_richness[[1]],full_richness[[3]],full_richness[[5]],full_richness[[7]]),
                  c(full_richness[[2]],full_richness[[4]],full_richness[[6]],full_richness[[8]])))
```

It appears that this difference in richness is driven by an outlier...

```{r,fig.height=4, fig.width=10}
stripchart(list(c(full_richness[[1]],full_richness[[3]],full_richness[[5]],full_richness[[7]]),
                c(full_richness[[2]],full_richness[[4]],full_richness[[6]],full_richness[[8]])),
           method='jitter',jitter=.1,vertical=F,col='navyblue',pch=16,cex=1.2,main="Richness by Patient Group",
           ylab="Patient Group", xlab="Richness")
```

Recomputing after removing richness outlier in mBAL 

The mean + 3SD of mBAL richness
```{r,fig.height=4, fig.width=10}
original_mbal_richness <- (unlist(list(c(full_richness[[1]],full_richness[[3]],full_richness[[5]],full_richness[7]))))
outlier_threshold <- mean(original_mbal_richness) + sd(original_mbal_richness)*3
new_values <- original_mbal_richness[original_mbal_richness < outlier_threshold]
print(wilcox.test(new_values,c(full_richness[[2]],full_richness[[4]],full_richness[[6]],full_richness[[8]])))
stripchart(list(new_values,c(full_richness[[2]],full_richness[[4]],full_richness[[6]],full_richness[[8]])),method='jitter',jitter=.1,vertical=F,col='navyblue',pch=16,cex=1.2,main="Richness by Patient Group",ylab="Patient Group", xlab="Richness")
```

Overall mBAL richness summary
```{r, echo = FALSE}
kable(t(as.matrix(summary(new_values))),"markdown") %>%
  kable_styling(bootstrap_options = c("striped","hover","condensed","responsive"),full_width=F)
```

Overall TA richness summary
```{r, echo = FALSE}
kable(t(as.matrix(summary(c(full_richness[[2]],full_richness[[4]],full_richness[[6]],full_richness[[8]])))),"markdown") %>%
  kable_styling(bootstrap_options = c("striped","hover","condensed","responsive"),full_width=F)
```

Comparing genus-level richness between PNA-pos and PNA-neg patients in mBAL 
```{r, echo = FALSE}
print(wilcox.test(full_richness[[1]],full_richness[[7]]))
```

Comparing genus-level richness between PNA-pos and PNA-neg patients in TA
```{r, echo = FALSE}
print(wilcox.test(full_richness[[2]],full_richness[[8]]))
```


```{r, results = "hide", include =FALSE}
if(generate_pdfs){
  pdf('./output/genus.richness.pdf',height=5, width=5)
}
names(full_richness) <- c("Group 1\nMBAL","Group 1\nTA", "Group 2\nMBAL","Group 2\nTA", "Group 3\nMBAL","Group 3\nTA", "Group 4\nMBAL","Group 4\nTA")
boxplot(full_richness,cex.axis=.5,main="Richness by Sample Type and Patient Group",xlab="Patient Group", ylab="Richness",cex.main=.8)
stripchart(full_richness,method='jitter',jitter=.1,vertical=T,col='navyblue',pch=16,cex=1,add=T)
# dev.off()
```

```{r, echo=FALSE, include=FALSE}
print("mean/sd richness mBAL Group 1:")
print(mean(unlist(full_richness[[1]])))
print(sd(unlist(full_richness[[1]])))
print("mean/sd richness TA Group 1:")
print(mean(unlist(full_richness[[2]])))
print(sd(unlist(full_richness[[2]])))
print("mean/sd richness mBAL Group 4:")
print(mean(unlist(full_richness[[7]])))
print(sd(unlist(full_richness[[7]])))
print("mean/sd richness TA Group 4:")
print(mean(unlist(full_richness[[8]])))
print(sd(unlist(full_richness[[8]])))
```

***

### Evaluate the presence/absence of clinically-confirmed pathogens in Group 1 Patients 

```{r, fig.width=10, fig.height=6, results = FALSE, echo = FALSE}

relative_numbers <- FALSE
column_names <- c()
IDs <- c()
overlap_both <- c()
rank_list_mbal <- c()
rank_list_ta <- c()
cummulative_pathos <- c()
sample_ids <- c()

plot(c(.5,2.5),c(0,1),main="Proportion of Sample",ylab="Proportion of Sample",xlab="TA (1) v. mBAL (2)")
for( i in rownames(metadata)){
  mbal_filename <- as.character(metadata[rownames(metadata) == i,]$mBAL_DNAseq_filename)
  ta_filename <- as.character(metadata[rownames(metadata) == i,]$TA_DNAseq_filename)
  
  
  if(!is.na(metadata[rownames(metadata) == i,]$effective_group)){
    if(metadata[rownames(metadata) == i,]$effective_group == 1){
      print(mbal_filename)
      print(ta_filename)
      
      # sort mBAL samples and get names
      mbal <- sort(group1.MBAL.microbes.norm[,c(mbal_filename)],decreasing=TRUE)
      mbal_names <- names(mbal[mbal>.01])
      
      if(ta_filename %in% colnames(group1.TA.microbes.norm)){
        
        IDs <- c(IDs,rownames(metadata[rownames(metadata) == i,]))
        
        # sort TA samples and get names
        ta <- sort(group1.TA.microbes.norm[,c(ta_filename)],decreasing=TRUE)
        ta_names <- names(ta[ta>.01])

        # get the pathogens        
        pathogens <- as.character(metadata[rownames(metadata) == i,]$microbe)
        patholist <- strsplit(pathogens,',')[[1]]
        print(patholist)
        
        rank_mbal <- match(patholist,mbal_names) 
        rank_ta <- match(patholist,ta_names)
        print(paste("rank mBAL",rank_mbal))
        print(paste("rank TA",rank_ta))
        
        rank_list_mbal <- c(rank_list_mbal,rank_mbal)
        rank_list_ta <- c(rank_list_ta, rank_ta)
        cummulative_pathos <- c(cummulative_pathos, patholist)
        sample_ids <- c(sample_ids, rep(strsplit(mbal_filename,"[.]")[[1]][2],length(patholist)))
        
        #move these lines here if you want to only include the pathogens in the barplot
        mbal_names <- mbal_names[mbal_names %in% patholist]
        ta_names <- ta_names[ta_names %in% patholist]
        
        shared <- intersect(as.vector(mbal_names), as.vector(ta_names))
        mbal_specific <- setdiff(mbal_names, ta_names) #gives the number that are present only in mbal_set
        ta_specific <-  setdiff(ta_names, mbal_names)  #gives the number that are present only in ta_set
        
        #print(intersect(shared,patholist))
        overlap_both<-c(overlap_both, length(intersect(shared,patholist)))
        if(length(intersect(shared,patholist)) == length(patholist)){
          #print("ALL FOUND")
          for(p in patholist){
            points(c(1,2),c(ta[p],mbal[p]),type="o")
            text(c(1,2),c(ta[p],mbal[p]),p,cex=.8)
          }
        }else{
          for(p in patholist){
            if(p %in% ta_names){
              points(c(1),c(ta[p]),pch=4)
            }
            if(p %in% mbal_names){
              points(c(2),c(mbal[p]),pch=4)
            }
          }
        }
        
        if(length(relative_numbers) < 3){
          relative_numbers <- c(length(ta_specific), length(shared), length(mbal_specific))
        }else{
          relative_numbers <- cbind(relative_numbers, c(length(ta_specific), length(shared), length(mbal_specific)) )
        }
        cbind(relative_numbers, c(ta_specific, shared, mbal_specific))  
        
        relative_numbers <- as.matrix(relative_numbers)
        rownames(relative_numbers) <- c("TA", "BOTH", "MBAL")
        column_names <- c(column_names, mbal_filename[1])
        colnames(relative_numbers) <- column_names    
     
      }
    }    
  }
}
```


```{r, fig.height=4, fig.width=10, echo = FALSE}
#Output plots 
#generate_pdfs = TRUE
if(generate_pdfs){
  pdf('./output/pathogens_identified.pdf',width=8, height=6)
}

par(mar=c(10,3,2,2))
colnames(relative_numbers) <- IDs
barplot(relative_numbers, col=c("gray80","navyblue","gray50"),main="Microbes Identified in DNA-seq by TA and/or mBAL",legend=TRUE,cex.main=.8,ylab="# of Pathogens Cultured",xlab="Patient ID",cex.axis=.6, las=2)
# dev.off()

write.csv(as.data.frame(cbind(sample_ids,cummulative_pathos,rank_list_mbal,rank_list_ta)),'./output/ranks_by_sample_type.tsv',quote=F)
kable(as.data.frame(cbind(sample_ids,cummulative_pathos,rank_list_mbal,rank_list_ta)), "markdown") %>% 
  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

*** 

### Analysis of a. priori selected oropharyngeal microbe contaiminants

Oropharyngeal contaminant microbes were selected a. priori from this paper: 
Dickson, R. P. et al. Bacterial Topography of the Healthy Human Lower Respiratory Tract. mBio 8, (2017).
In which they describe bacterial taxa identified by 16S amplicon sequencing in the airway from 1. Oral rinse, 2. Protected specimen brush, 3. Bronchoalveolar lavage.

```{r, fig.width=12, fig.height=4, echo = FALSE}

colorset1 <- c('#0F726F','#B01833')

if(generate_pdfs){
  pdf('./output/Apriori_genera.pdf',width=18, height=5)
}

par(mfrow=c(1,6))
stripchart(list(all_microbe_data[c("Prevotella ( 838 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Prevotella ( 838 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]),vertical=TRUE,jitter=.1,method='jitter',pch=16,cex=2,col=colorset1,main="Prevotella ( 838 )", ylim=c(0,1))
text(c(2),c(.8),round(wilcox.test(all_microbe_data[c("Prevotella ( 838 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Prevotella ( 838 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value,3))
#print(wilcox.test(all_microbe_data[c("Prevotella ( 838 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Prevotella ( 838 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value)

stripchart(list(all_microbe_data[c("Veillonella ( 29465 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Veillonella ( 29465 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]),vertical=TRUE,jitter=.1,method='jitter',pch=16,cex=2,col=colorset1,main="Veillonella ( 29465 )", ylim=c(0,1))
text(c(2),c(.8),round(wilcox.test(all_microbe_data[c("Veillonella ( 29465 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Veillonella ( 29465 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value,3))
#print(wilcox.test(all_microbe_data[c("Veillonella ( 29465 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Veillonella ( 29465 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value)

stripchart(list(all_microbe_data[c("Streptococcus ( 1301 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Streptococcus ( 1301 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]),vertical=TRUE,jitter=.1,method='jitter',pch=16,cex=2,col=colorset1,main="Streptococcus ( 1301 )", ylim=c(0,1))
text(c(2),c(.8),round(wilcox.test(all_microbe_data[c("Streptococcus ( 1301 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Streptococcus ( 1301 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value,3))

stripchart(list(all_microbe_data[c("Fusobacterium ( 848 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Fusobacterium ( 848 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]),vertical=TRUE,jitter=.1,method='jitter',pch=16,cex=2,col=colorset1,main="Fusobacterium ( 848 )", ylim=c(0,1))
text(c(2),c(.8),round(wilcox.test(all_microbe_data[c("Fusobacterium ( 848 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Fusobacterium ( 848 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value,3))

stripchart(list(all_microbe_data[c("Rothia ( 32207 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Rothia ( 32207 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]),vertical=TRUE,jitter=.1,method='jitter',pch=16,cex=2,col=colorset1,main="Rothia ( 32207 )", ylim=c(0,1))
text(c(2),c(.8),round(wilcox.test(all_microbe_data[c("Rothia ( 32207 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Rothia ( 32207 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value,3))

stripchart(list(all_microbe_data[c("Neisseria ( 482 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Neisseria ( 482 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]),vertical=TRUE,jitter=.1,method='jitter',pch=16,cex=2,col=colorset1,main="Neisseria ( 482 )", ylim=c(0,1))
text(c(2),c(.8),round(wilcox.test(all_microbe_data[c("Neisseria ( 482 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data[c("Neisseria ( 482 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value,3))
```

Summary of Microbial Abundances in Tracheal Aspriate Samples:
```{r, echo=FALSE}
kable(as.matrix(summary(t(all_microbe_data[c("Prevotella ( 838 )","Veillonella ( 29465 )","Streptococcus ( 1301 )","Fusobacterium ( 848 )","Rothia ( 32207 )","Neisseria ( 482 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))]))),"markdown") #%>% 
#  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

Summary of Microbial Abundances in mini-BAL Samples:
```{r, echo=FALSE}
kable(as.matrix(summary(t(all_microbe_data[c("Prevotella ( 838 )","Veillonella ( 29465 )","Streptococcus ( 1301 )","Fusobacterium ( 848 )","Rothia ( 32207 )","Neisseria ( 482 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]))),"markdown")# %>% 
#  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

***

```{r, fig.width=12, fig.height=4, echo = FALSE, include=FALSE}
# ### Analysis of a. priori oropharyngeal microbial contaminants 
# **(without removing things with < 1% abundance)**
# 
# Note: this analysis was not included in the manuscript for the sake of consistency with the other analyses.

all_microbe_data_orig <- as.matrix(all_microbe_data_orig)

par(mfrow=c(1,6))
stripchart(list(all_microbe_data_orig[c("Prevotella ( 838 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Prevotella ( 838 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]),vertical=TRUE,jitter=.1,method='jitter',pch=16,cex=2,col=colorset1,main="Prevotella ( 838 )", ylim=c(0,1))
text(c(2),c(.8),round(wilcox.test(all_microbe_data_orig[c("Prevotella ( 838 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Prevotella ( 838 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value,3))
#print(wilcox.test(all_microbe_data_orig[c("Prevotella ( 838 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Prevotella ( 838 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value)

stripchart(list(all_microbe_data_orig[c("Veillonella ( 29465 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Veillonella ( 29465 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]),vertical=TRUE,jitter=.1,method='jitter',pch=16,cex=2,col=colorset1,main="Veillonella ( 29465 )", ylim=c(0,1))
text(c(2),c(.8),round(wilcox.test(all_microbe_data_orig[c("Veillonella ( 29465 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Veillonella ( 29465 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value,3))
#print(wilcox.test(all_microbe_data_orig[c("Veillonella ( 29465 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Veillonella ( 29465 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value)

stripchart(list(all_microbe_data_orig[c("Streptococcus ( 1301 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Streptococcus ( 1301 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]),vertical=TRUE,jitter=.1,method='jitter',pch=16,cex=2,col=colorset1,main="Streptococcus ( 1301 )", ylim=c(0,1))
text(c(2),c(.8),round(wilcox.test(all_microbe_data_orig[c("Streptococcus ( 1301 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Streptococcus ( 1301 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value,3))

stripchart(list(all_microbe_data_orig[c("Fusobacterium ( 848 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Fusobacterium ( 848 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]),vertical=TRUE,jitter=.1,method='jitter',pch=16,cex=2,col=colorset1,main="Fusobacterium ( 848 )", ylim=c(0,1))
text(c(2),c(.8),round(wilcox.test(all_microbe_data_orig[c("Fusobacterium ( 848 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Fusobacterium ( 848 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value,3))

stripchart(list(all_microbe_data_orig[c("Rothia ( 32207 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Rothia ( 32207 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]),vertical=TRUE,jitter=.1,method='jitter',pch=16,cex=2,col=colorset1,main="Rothia ( 32207 )", ylim=c(0,1))
text(c(2),c(.8),round(wilcox.test(all_microbe_data_orig[c("Rothia ( 32207 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Rothia ( 32207 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value,3))

stripchart(list(all_microbe_data_orig[c("Neisseria ( 482 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Neisseria ( 482 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]),vertical=TRUE,jitter=.1,method='jitter',pch=16,cex=2,col=colorset1,main="Neisseria ( 482 )", ylim=c(0,1))
text(c(2),c(.8),round(wilcox.test(all_microbe_data_orig[c("Neisseria ( 482 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))],all_microbe_data_orig[c("Neisseria ( 482 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))])$p.value,3))
```

```{r, echo=FALSE, include=FALSE}
# Summary of Microbial Abundances in Tracheal Aspriate Samples:
kable(as.matrix(summary(t(all_microbe_data_orig[c("Prevotella ( 838 )","Veillonella ( 29465 )","Streptococcus ( 1301 )","Fusobacterium ( 848 )","Rothia ( 32207 )","Neisseria ( 482 )"),as.character(unlist(metadata[,c("TA_DNAseq_filename")]))]))),"markdown") #%>% 
#  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

```{r, echo=FALSE, include=FALSE}
# Summary of Microbial Abundances in mini-BAL Samples:
kable(as.matrix(summary(t(all_microbe_data_orig[c("Prevotella ( 838 )","Veillonella ( 29465 )","Streptococcus ( 1301 )","Fusobacterium ( 848 )","Rothia ( 32207 )","Neisseria ( 482 )"),as.character(unlist(metadata[,c("mBAL_DNAseq_filename")]))]))),"markdown") # %>% 
#  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```




























